drivers/dai/intel/alh/alh.c

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <spinlock.h>
+#include <devicetree.h>
+#include <logging/log.h>
+
+#define DT_DRV_COMPAT intel_alh_dai
+#define LOG_DOMAIN dai_intel_alh
+
+LOG_MODULE_REGISTER(LOG_DOMAIN);
+
+#include "alh.h"
+
+/* Digital Audio interface formatting */
+static int dai_alh_set_config_tplg(struct dai_intel_alh *dp, const struct dai_config *common_config,
+				   const void *spec_config)
+{
+	struct dai_intel_alh_pdata *alh = dai_get_drvdata(dp);
+	const struct dai_intel_ipc3_alh_params *config = spec_config;
+
+	if (config->channels || config->rate) {
+		alh->params.channels = config->channels;
+		alh->params.rate = config->rate;
+		LOG_INF("%s channels %d rate %d", __func__, config->channels, config->rate);
+	}
+
+	alh->params.stream_id = config->stream_id;
+
+	return 0;
+}
+
+static int dai_alh_set_config_blob(struct dai_intel_alh *dp, const void *spec_config)
+{
+	struct dai_intel_alh_pdata *alh = dai_get_drvdata(dp);
+	const struct dai_intel_ipc4_alh_configuration_blob *blob = spec_config;
+	const struct ipc4_alh_multi_gtw_cfg *alh_cfg = &blob->alh_cfg;
+
+	alh->params.channels = ALH_CHANNELS_DEFAULT;
+	alh->params.rate = ALH_RATE_DEFAULT;
+	/* the LSB 8bits are for stream id */
+	alh->params.stream_id = alh_cfg->mapping[0].alh_id & 0xff;
+
+	return 0;
+}
+
+static int dai_alh_trigger(const struct device *dev, enum dai_dir dir,
+			   enum dai_trigger_cmd cmd)
+{
+	LOG_INF("%s cmd %d", __func__, cmd);
+
+	return 0;
+}
+
+static void alh_claim_ownership(void)
+{
+#if !CONFIG_INTEL_ADSP_CAVS
```

Comment:
Agree ... It should be abstracted for the feature and not an specific hw. Than the symbol is selected when supported.

Reply:
This is temporary because it should be "#if CONFIG_INTEL_ADSP_ACE" but _ACE doesnt exist on upstream yet so for now I just used !CAVS with planning to change in the future

But I can also change to CONFIG_DAI_ALH_HAS_OWNERSHIP

Reply:
@abonislawski lets add ALH_OWNERSHIP feature flag and use this instead of platform, otherwise we will have to update the list of platforms that actually support it.

Reply:
@mwasko its done with the latest update